test/sqllogictest/column_knowledge.slt

```diff
@@ -77,26 +77,22 @@ EXPLAIN SELECT * FROM t1 LEFT JOIN t2 ON (t1.f1 = t2.f1) WHERE t1.f1 = 123;
 | | demand = (#0..#3)
 
 %4 =
-| Get %0 (l0)
+| Get %3 (l1)
+| Distinct group=()
```

Comment:
What does group on ```()``` mean?  It was rarely seen before and seem to imply grouping over all columns of the input.

With this branch, one can get chained Distincts like this:

```
materialize=>  explain  select distinct 234 from (select distinct f1 from t1) ;
          Optimized Plan          
----------------------------------
 %0 =                            +
 | Get materialize.public.t1 (u1)+
 | Distinct group=(#0)           +
 | Distinct group=()             +
 | Map 234                       +
```

Reply:
I guess `Distinct group=()` means _one row at most_. There should be another rule remove nested Distinct operator when the outer one is more restrictive than the inner one.

Reply:
Created issue for removing nested operator at #6543.

`select distinct 234 from (select distinct f1 from t1) ;` will return `234` if t1 has any rows. If t1 has no rows, it will return no rows.

So `Distinct group()` means "return zero rows if input has zero rows, else return one row with zero columns."